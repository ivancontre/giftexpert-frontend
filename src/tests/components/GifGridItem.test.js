import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import GifGridItem from '../../components/GifGridItem';

describe('Pruebas sobre GifGridItem', () => {

    const title = 'Este es un título';
    const url = 'https_//localhost/algo.png';
    const wrapper = shallow(<GifGridItem title={ title } url={ url } />);

    test('debe de mostrar el componente correctamente ', () => {        
        expect(wrapper).toMatchSnapshot();
    })   
    
    test('debe de tener un párrafo con el title  ', () => {
        const textoTitulo = wrapper.find('p').text().trim();
        expect(title).toBe(textoTitulo);
    })

    test('debe de tener la imagen igual al url y al alt de los props', () => {
        const img = wrapper.find('img');
        expect(img.prop('src')).toBe(url);
        expect(img.prop('alt')).toBe(title);
    })

    test('debe de tener la clase "animate__fadeIn" ', () => {
        const div = wrapper.find('div');
        console.log(div.prop('className'))
        expect(div.prop('className').includes('animate__fadeIn')).toBe(true);
    })
    
    
    
})
