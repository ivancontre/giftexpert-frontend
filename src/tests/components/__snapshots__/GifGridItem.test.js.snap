// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas sobre GifGridItem debe de mostrar el componente correctamente  1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Este es un título"
    src="https_//localhost/algo.png"
  />
  <p>
    Este es un título
     
  </p>
</div>
`;
